### PR TITLE
OCPBUGS-36481: Fix Hypershift dump for non-OpenShift Management Clusters

### PR DESCRIPTION
**Pull request by fxierh**
When the MC is not OpenShift, some `oc adm inspect` commands invoked by hypershift dump error out as the server does not recognize OpenShift resources like ImageStream. As a result, many GVKs are missing in the dump. 

Only dump registered OpenShift GVKs in this case.

### CODE REVIEW

**Comment by fxierh** [comment]
Dump an AKS MC without this PR:
```bash
$ hypershift dump cluster --artifact-dir . --name=b5bbad7d87e75aa73461
$ tree -d
.
├── cluster-scoped-resources
│   └── core
│       └── nodes
├── namespaces
│   ├── clusters
│   │   └── hypershift.openshift.io
│   │       ├── hostedclusters
│   │       └── nodepools
│   ├── clusters-b5bbad7d87e75aa73461
│   │   └── core
│   │       └── pods
│   │           └── logs
│   └── hypershift
│       └── core
│           └── pods
│               └── logs
└── network_logs
```

Dump an AKS MC with this PR:
```bash
$ hypershift dump cluster --artifact-dir . --name=b5bbad7d87e75aa73461
$ tree -d
.
├── cluster-scoped-resources
│   └── core
│       └── nodes
├── namespaces
│   ├── clusters
│   │   ├── core
│   │   │   └── configmaps
│   │   └── hypershift.openshift.io
│   │       ├── hostedclusters
│   │       └── nodepools
│   ├── clusters-b5bbad7d87e75aa73461
│   │   ├── apps
│   │   │   ├── deployments
│   │   │   ├── replicasets
│   │   │   └── statefulsets
│   │   ├── cluster.x-k8s.io
│   │   │   ├── clusters
│   │   │   ├── machinedeployments
│   │   │   ├── machines
│   │   │   └── machinesets
│   │   ├── core
│   │   │   ├── configmaps
│   │   │   ├── endpoints
│   │   │   ├── events
│   │   │   ├── persistentvolumeclaims
│   │   │   ├── pods
│   │   │   │   └── logs
│   │   │   └── services
│   │   ├── hypershift.openshift.io
│   │   │   └── hostedcontrolplanes
│   │   ├── networking.k8s.io
│   │   │   └── networkpolicies
│   │   ├── policy
│   │   │   └── poddisruptionbudgets
│   │   └── route.openshift.io
│   │       └── routes
│   └── hypershift
│       ├── apps
│       │   ├── deployments
│       │   └── replicasets
│       └── core
│           ├── configmaps
│           ├── endpoints
│           ├── pods
│           │   └── logs
│           └── services
└── network_logs

48 directories
```

File: cmd/cluster/core/dump.go [hunk 1]
```diff
@@ -765,3 +788,19 @@ func shouldDumpKubevirt(nodePools []*hyperv1.NodePool) ([]kubevirtExtCluster, bo
 
 	return kubevirtInExternalInfras, localKubevirtInUse
 }
+
+func isResourceRegistered(discoveryClient discovery.DiscoveryInterface, gvk schema.GroupVersionKind) (bool, error) {
```
Comment (enxebre) [inline cmd/cluster/core/dump.go:792]: can we unit test this? see e.g. https://github.com/kubernetes/autoscaler/blob/4d0511363eeeff657119797dd8d26e851dcc3459/cluster-autoscaler/cloudprovider/clusterapi/clusterapi_controller_test.go#L123C2-L123C17

Comment (fxierh) [reply cmd/cluster/core/dump.go:792]: @enxebre Yup. Thanks for the example. Updated.

**Comment by enxebre** [comment]
Thanks!
/label tide/merge-method-squash
/approve

**Comment by fxierh** [comment]
Cherry-picked to version 4.16 as it is the minimum version that supports the AKS MC.

**Pull request merged**